e2e: -count=N does not work because channel namespace is terminating

Trying to run the same tests multiple times for evaluating test flakiness breaks because the channel is deleted during cleanup.

Ideally the flow will be:

1. global setup before all tests
2. run one or more tests count times
3. global teardown before existing

I'm not sure this is possible with -count=N.

If not possible and the channel need to be deleted, we mitigate this in several ways:

- Wait until the channel and the namespace are deleted during cleanup, so installing the channel again will not fail.
- Add a flag to skip the cleanup, to be used for special cases like -count=N
- Add flag to skip adding/removing the channel, it can be added manually
- Implement -count=N internally so we can control setup and teardown 

## Example run

```
% ./run.sh --config ../e2e.yaml -run TestSuites/Exhaustive/Deploy-cephfs/Subscr -count=2
Running tests...
2024-10-28T19:48:18.586+0200	INFO	e2e/main_test.go:41	Global setting	{"Timeout": 600}
2024-10-28T19:48:18.587+0200	INFO	e2e/main_test.go:42	Global setting	{"Retry Interval": 30}
=== RUN   TestSuites
2024-10-28T19:48:18.587+0200	INFO	e2e/main_test.go:57	TestSuites
=== RUN   TestSuites/Exhaustive
=== PAUSE TestSuites/Exhaustive
=== CONT  TestSuites/Exhaustive
2024-10-28T19:48:18.618+0200	INFO	util/crud.go:77	channel ramen-gitops is created
=== RUN   TestSuites/Exhaustive/Deploy-cephfs
=== PAUSE TestSuites/Exhaustive/Deploy-cephfs
=== RUN   TestSuites/Exhaustive/Deploy-cephfs#01
=== PAUSE TestSuites/Exhaustive/Deploy-cephfs#01
=== RUN   TestSuites/Exhaustive/Deploy-cephfs#02
=== PAUSE TestSuites/Exhaustive/Deploy-cephfs#02
=== CONT  TestSuites/Exhaustive/Deploy-cephfs
=== CONT  TestSuites/Exhaustive/Deploy-cephfs#02
=== CONT  TestSuites/Exhaustive/Deploy-cephfs#01
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr
=== PAUSE TestSuites/Exhaustive/Deploy-cephfs/Subscr
=== CONT  TestSuites/Exhaustive/Deploy-cephfs/Subscr
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Deploy
2024-10-28T19:48:18.619+0200	INFO	deployers/subscription.go:31	enter Deploy subscr-deploy-cephfs-busybox
2024-10-28T19:48:23.750+0200	INFO	deployers/retry.go:32	subscription subscr-deploy-cephfs-busybox phase is Propagated
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Enable
2024-10-28T19:48:23.751+0200	INFO	dractions/actions.go:39	enter EnableProtection subscr-deploy-cephfs-busybox
2024-10-28T19:48:23.755+0200	INFO	dractions/actions.go:52	got clusterName dr1 from subscr-deploy-cephfs-busybox-decision-1
2024-10-28T19:48:23.755+0200	INFO	dractions/actions.go:54	update placement subscr-deploy-cephfs-busybox annotation
2024-10-28T19:48:23.762+0200	INFO	dractions/actions.go:73	create drpc subscr-deploy-cephfs-busybox
2024-10-28T19:51:23.842+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Failover
2024-10-28T19:51:23.843+0200	INFO	dractions/actions.go:115	enter Failover subscr-deploy-cephfs-busybox
2024-10-28T19:51:23.851+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
2024-10-28T19:51:23.861+0200	INFO	dractions/actions.go:175	update drpc subscr-deploy-cephfs-busybox failover to dr2
2024-10-28T19:52:28.884+0200	INFO	dractions/retry.go:114	drpc subscr-deploy-cephfs-busybox phase is FailedOver
2024-10-28T19:53:58.901+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Relocate
2024-10-28T19:53:58.901+0200	INFO	dractions/actions.go:130	enter Relocate subscr-deploy-cephfs-busybox
2024-10-28T19:53:58.903+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
2024-10-28T19:53:58.909+0200	INFO	dractions/actions.go:175	update drpc subscr-deploy-cephfs-busybox relocate to dr1
2024-10-28T19:56:33.939+0200	INFO	dractions/retry.go:114	drpc subscr-deploy-cephfs-busybox phase is Relocated
2024-10-28T19:57:03.947+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Disable
2024-10-28T19:57:03.947+0200	INFO	dractions/actions.go:99	enter DisableProtection subscr-deploy-cephfs-busybox
2024-10-28T19:57:03.947+0200	INFO	dractions/actions.go:104	delete drpc subscr-deploy-cephfs-busybox
2024-10-28T19:57:38.966+0200	INFO	dractions/retry.go:183	drpc subscr-deploy-cephfs-busybox is deleted
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Undeploy
2024-10-28T19:57:38.966+0200	INFO	deployers/subscription.go:62	enter Undeploy subscr-deploy-cephfs-busybox
2024-10-28T19:57:39.009+0200	INFO	util/crud.go:99	channel ramen-gitops is deleted
--- PASS: TestSuites (0.00s)
    --- PASS: TestSuites/Exhaustive (0.06s)
        --- PASS: TestSuites/Exhaustive/Deploy-cephfs#02 (0.00s)
        --- PASS: TestSuites/Exhaustive/Deploy-cephfs#01 (0.00s)
        --- PASS: TestSuites/Exhaustive/Deploy-cephfs (0.00s)
            --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr (560.38s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Deploy (5.13s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Enable (180.09s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Failover (155.06s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Relocate (185.05s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Disable (35.01s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Undeploy (0.03s)
=== RUN   TestSuites
2024-10-28T19:57:39.027+0200	INFO	e2e/main_test.go:57	TestSuites
=== RUN   TestSuites/Exhaustive
=== PAUSE TestSuites/Exhaustive
=== CONT  TestSuites/Exhaustive
    main_test.go:64: failed to ensure channel: channels.apps.open-cluster-management.io "ramen-gitops" is forbidden: unable to create new content in namespace ramen-samples because it is being terminated
--- FAIL: TestSuites (0.00s)
    --- FAIL: TestSuites/Exhaustive (0.05s)
FAIL
exit status 1
FAIL	github.com/ramendr/ramen/e2e	561.243s
```

## Workaround

Run the entire test multiple times instead of using -count:

```
% for i in $(seq 10); do ./run.sh -run TestSuites/Exhaustive/Deploy-cephfs/Subscr; done
```

## Tasks

- [ ] Understand why the channel is not delete once at the end of all tests when running with -count=N
- [ ] Test way to mitigate this issue

Tested the workaround - it does not solve the issue - starting the test still fails sometimes because the channel namespace is still deleting when we start the tests. So the root cause is not waiting until the namespace is deleted during cleanup.

Example run:

```
% for i in $(seq 5); do ./run.sh --config ../e2e.yaml -run TestSuites/Exhaustive/Deploy-cephfs/Subscr; done
Running tests...
2024-10-28T21:07:52.744+0200	INFO	e2e/main_test.go:41	Global setting	{"Timeout": 600}
2024-10-28T21:07:52.744+0200	INFO	e2e/main_test.go:42	Global setting	{"Retry Interval": 30}
=== RUN   TestSuites
2024-10-28T21:07:52.745+0200	INFO	e2e/main_test.go:57	TestSuites
=== RUN   TestSuites/Exhaustive
=== PAUSE TestSuites/Exhaustive
=== CONT  TestSuites/Exhaustive
2024-10-28T21:07:52.771+0200	INFO	util/crud.go:77	channel ramen-gitops is created
=== RUN   TestSuites/Exhaustive/Deploy-cephfs
=== PAUSE TestSuites/Exhaustive/Deploy-cephfs
=== RUN   TestSuites/Exhaustive/Deploy-cephfs#01
=== PAUSE TestSuites/Exhaustive/Deploy-cephfs#01
=== RUN   TestSuites/Exhaustive/Deploy-cephfs#02
=== PAUSE TestSuites/Exhaustive/Deploy-cephfs#02
=== CONT  TestSuites/Exhaustive/Deploy-cephfs
=== CONT  TestSuites/Exhaustive/Deploy-cephfs#02
=== CONT  TestSuites/Exhaustive/Deploy-cephfs#01
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr
=== PAUSE TestSuites/Exhaustive/Deploy-cephfs/Subscr
=== CONT  TestSuites/Exhaustive/Deploy-cephfs/Subscr
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Deploy
2024-10-28T21:07:52.772+0200	INFO	deployers/subscription.go:31	enter Deploy subscr-deploy-cephfs-busybox
2024-10-28T21:07:57.908+0200	INFO	deployers/retry.go:32	subscription subscr-deploy-cephfs-busybox phase is Propagated
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Enable
2024-10-28T21:07:57.908+0200	INFO	dractions/actions.go:39	enter EnableProtection subscr-deploy-cephfs-busybox
2024-10-28T21:07:57.913+0200	INFO	dractions/actions.go:52	got clusterName dr1 from subscr-deploy-cephfs-busybox-decision-1
2024-10-28T21:07:57.913+0200	INFO	dractions/actions.go:54	update placement subscr-deploy-cephfs-busybox annotation
2024-10-28T21:07:57.920+0200	INFO	dractions/actions.go:73	create drpc subscr-deploy-cephfs-busybox
2024-10-28T21:11:28.019+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Failover
2024-10-28T21:11:28.019+0200	INFO	dractions/actions.go:115	enter Failover subscr-deploy-cephfs-busybox
2024-10-28T21:11:28.022+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
2024-10-28T21:11:28.027+0200	INFO	dractions/actions.go:175	update drpc subscr-deploy-cephfs-busybox failover to dr2
2024-10-28T21:12:03.047+0200	INFO	dractions/retry.go:114	drpc subscr-deploy-cephfs-busybox phase is FailedOver
2024-10-28T21:13:33.088+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Relocate
2024-10-28T21:13:33.088+0200	INFO	dractions/actions.go:130	enter Relocate subscr-deploy-cephfs-busybox
2024-10-28T21:13:33.091+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
2024-10-28T21:13:33.099+0200	INFO	dractions/actions.go:175	update drpc subscr-deploy-cephfs-busybox relocate to dr1
2024-10-28T21:16:08.133+0200	INFO	dractions/retry.go:114	drpc subscr-deploy-cephfs-busybox phase is Relocated
2024-10-28T21:16:38.139+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Disable
2024-10-28T21:16:38.139+0200	INFO	dractions/actions.go:99	enter DisableProtection subscr-deploy-cephfs-busybox
2024-10-28T21:16:38.139+0200	INFO	dractions/actions.go:104	delete drpc subscr-deploy-cephfs-busybox
2024-10-28T21:17:13.153+0200	INFO	dractions/retry.go:183	drpc subscr-deploy-cephfs-busybox is deleted
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Undeploy
2024-10-28T21:17:13.153+0200	INFO	deployers/subscription.go:62	enter Undeploy subscr-deploy-cephfs-busybox
2024-10-28T21:17:13.229+0200	INFO	util/crud.go:99	channel ramen-gitops is deleted
--- PASS: TestSuites (0.00s)
    --- PASS: TestSuites/Exhaustive (0.05s)
        --- PASS: TestSuites/Exhaustive/Deploy-cephfs#01 (0.00s)
        --- PASS: TestSuites/Exhaustive/Deploy-cephfs#02 (0.00s)
        --- PASS: TestSuites/Exhaustive/Deploy-cephfs (0.00s)
            --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr (560.45s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Deploy (5.14s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Enable (210.11s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Failover (125.07s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Relocate (185.05s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Disable (35.01s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Undeploy (0.06s)
PASS
ok  	github.com/ramendr/ramen/e2e	561.167s
Running tests...
2024-10-28T21:17:17.191+0200	INFO	e2e/main_test.go:41	Global setting	{"Timeout": 600}
2024-10-28T21:17:17.192+0200	INFO	e2e/main_test.go:42	Global setting	{"Retry Interval": 30}
=== RUN   TestSuites
2024-10-28T21:17:17.192+0200	INFO	e2e/main_test.go:57	TestSuites
=== RUN   TestSuites/Exhaustive
=== PAUSE TestSuites/Exhaustive
=== CONT  TestSuites/Exhaustive
    main_test.go:64: failed to ensure channel: channels.apps.open-cluster-management.io "ramen-gitops" is forbidden: unable to create new content in namespace ramen-samples because it is being terminated
--- FAIL: TestSuites (0.00s)
    --- FAIL: TestSuites/Exhaustive (0.02s)
FAIL
exit status 1
FAIL	github.com/ramendr/ramen/e2e	0.688s
Running tests...
2024-10-28T21:17:19.117+0200	INFO	e2e/main_test.go:41	Global setting	{"Timeout": 600}
2024-10-28T21:17:19.118+0200	INFO	e2e/main_test.go:42	Global setting	{"Retry Interval": 30}
=== RUN   TestSuites
2024-10-28T21:17:19.118+0200	INFO	e2e/main_test.go:57	TestSuites
=== RUN   TestSuites/Exhaustive
=== PAUSE TestSuites/Exhaustive
=== CONT  TestSuites/Exhaustive
2024-10-28T21:17:19.144+0200	INFO	util/crud.go:77	channel ramen-gitops is created
=== RUN   TestSuites/Exhaustive/Deploy-cephfs
=== PAUSE TestSuites/Exhaustive/Deploy-cephfs
=== RUN   TestSuites/Exhaustive/Deploy-cephfs#01
=== PAUSE TestSuites/Exhaustive/Deploy-cephfs#01
=== RUN   TestSuites/Exhaustive/Deploy-cephfs#02
=== PAUSE TestSuites/Exhaustive/Deploy-cephfs#02
=== CONT  TestSuites/Exhaustive/Deploy-cephfs
=== CONT  TestSuites/Exhaustive/Deploy-cephfs#02
=== CONT  TestSuites/Exhaustive/Deploy-cephfs#01
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr
=== PAUSE TestSuites/Exhaustive/Deploy-cephfs/Subscr
=== CONT  TestSuites/Exhaustive/Deploy-cephfs/Subscr
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Deploy
2024-10-28T21:17:19.145+0200	INFO	deployers/subscription.go:31	enter Deploy subscr-deploy-cephfs-busybox
2024-10-28T21:17:24.226+0200	INFO	deployers/retry.go:32	subscription subscr-deploy-cephfs-busybox phase is Propagated
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Enable
2024-10-28T21:17:24.226+0200	INFO	dractions/actions.go:39	enter EnableProtection subscr-deploy-cephfs-busybox
2024-10-28T21:17:24.230+0200	INFO	dractions/actions.go:52	got clusterName dr1 from subscr-deploy-cephfs-busybox-decision-1
2024-10-28T21:17:24.230+0200	INFO	dractions/actions.go:54	update placement subscr-deploy-cephfs-busybox annotation
2024-10-28T21:17:24.237+0200	INFO	dractions/actions.go:73	create drpc subscr-deploy-cephfs-busybox
2024-10-28T21:18:54.314+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Failover
2024-10-28T21:18:54.314+0200	INFO	dractions/actions.go:115	enter Failover subscr-deploy-cephfs-busybox
2024-10-28T21:18:54.318+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
2024-10-28T21:18:54.334+0200	INFO	dractions/actions.go:175	update drpc subscr-deploy-cephfs-busybox failover to dr2
2024-10-28T21:19:29.352+0200	INFO	dractions/retry.go:114	drpc subscr-deploy-cephfs-busybox phase is FailedOver
2024-10-28T21:21:29.369+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Relocate
2024-10-28T21:21:29.369+0200	INFO	dractions/actions.go:130	enter Relocate subscr-deploy-cephfs-busybox
2024-10-28T21:21:29.372+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
2024-10-28T21:21:29.377+0200	INFO	dractions/actions.go:175	update drpc subscr-deploy-cephfs-busybox relocate to dr1
2024-10-28T21:24:04.414+0200	INFO	dractions/retry.go:114	drpc subscr-deploy-cephfs-busybox phase is Relocated
2024-10-28T21:25:04.429+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Disable
2024-10-28T21:25:04.429+0200	INFO	dractions/actions.go:99	enter DisableProtection subscr-deploy-cephfs-busybox
2024-10-28T21:25:04.429+0200	INFO	dractions/actions.go:104	delete drpc subscr-deploy-cephfs-busybox
2024-10-28T21:25:39.451+0200	INFO	dractions/retry.go:183	drpc subscr-deploy-cephfs-busybox is deleted
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Undeploy
2024-10-28T21:25:39.451+0200	INFO	deployers/subscription.go:62	enter Undeploy subscr-deploy-cephfs-busybox
2024-10-28T21:25:39.516+0200	INFO	util/crud.go:99	channel ramen-gitops is deleted
--- PASS: TestSuites (0.00s)
    --- PASS: TestSuites/Exhaustive (0.05s)
        --- PASS: TestSuites/Exhaustive/Deploy-cephfs#02 (0.00s)
        --- PASS: TestSuites/Exhaustive/Deploy-cephfs#01 (0.00s)
        --- PASS: TestSuites/Exhaustive/Deploy-cephfs (0.00s)
            --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr (500.36s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Deploy (5.08s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Enable (90.09s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Failover (155.06s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Relocate (215.06s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Disable (35.02s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Undeploy (0.05s)
PASS
ok  	github.com/ramendr/ramen/e2e	501.027s
Running tests...
2024-10-28T21:25:43.577+0200	INFO	e2e/main_test.go:41	Global setting	{"Timeout": 600}
2024-10-28T21:25:43.578+0200	INFO	e2e/main_test.go:42	Global setting	{"Retry Interval": 30}
=== RUN   TestSuites
2024-10-28T21:25:43.578+0200	INFO	e2e/main_test.go:57	TestSuites
=== RUN   TestSuites/Exhaustive
=== PAUSE TestSuites/Exhaustive
=== CONT  TestSuites/Exhaustive
    main_test.go:64: failed to ensure channel: channels.apps.open-cluster-management.io "ramen-gitops" is forbidden: unable to create new content in namespace ramen-samples because it is being terminated
--- FAIL: TestSuites (0.00s)
    --- FAIL: TestSuites/Exhaustive (0.03s)
FAIL
exit status 1
FAIL	github.com/ramendr/ramen/e2e	0.683s
Running tests...
2024-10-28T21:25:45.516+0200	INFO	e2e/main_test.go:41	Global setting	{"Timeout": 600}
2024-10-28T21:25:45.517+0200	INFO	e2e/main_test.go:42	Global setting	{"Retry Interval": 30}
=== RUN   TestSuites
2024-10-28T21:25:45.517+0200	INFO	e2e/main_test.go:57	TestSuites
=== RUN   TestSuites/Exhaustive
=== PAUSE TestSuites/Exhaustive
=== CONT  TestSuites/Exhaustive
2024-10-28T21:25:45.544+0200	INFO	util/crud.go:77	channel ramen-gitops is created
=== RUN   TestSuites/Exhaustive/Deploy-cephfs
=== PAUSE TestSuites/Exhaustive/Deploy-cephfs
=== RUN   TestSuites/Exhaustive/Deploy-cephfs#01
=== PAUSE TestSuites/Exhaustive/Deploy-cephfs#01
=== RUN   TestSuites/Exhaustive/Deploy-cephfs#02
=== PAUSE TestSuites/Exhaustive/Deploy-cephfs#02
=== CONT  TestSuites/Exhaustive/Deploy-cephfs
=== CONT  TestSuites/Exhaustive/Deploy-cephfs#02
=== CONT  TestSuites/Exhaustive/Deploy-cephfs#01
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr
=== PAUSE TestSuites/Exhaustive/Deploy-cephfs/Subscr
=== CONT  TestSuites/Exhaustive/Deploy-cephfs/Subscr
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Deploy
2024-10-28T21:25:45.545+0200	INFO	deployers/subscription.go:31	enter Deploy subscr-deploy-cephfs-busybox
2024-10-28T21:25:50.595+0200	INFO	deployers/retry.go:32	subscription subscr-deploy-cephfs-busybox phase is Propagated
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Enable
2024-10-28T21:25:50.595+0200	INFO	dractions/actions.go:39	enter EnableProtection subscr-deploy-cephfs-busybox
2024-10-28T21:25:50.599+0200	INFO	dractions/actions.go:52	got clusterName dr1 from subscr-deploy-cephfs-busybox-decision-1
2024-10-28T21:25:50.599+0200	INFO	dractions/actions.go:54	update placement subscr-deploy-cephfs-busybox annotation
2024-10-28T21:25:50.609+0200	INFO	dractions/actions.go:73	create drpc subscr-deploy-cephfs-busybox
2024-10-28T21:27:20.679+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Failover
2024-10-28T21:27:20.680+0200	INFO	dractions/actions.go:115	enter Failover subscr-deploy-cephfs-busybox
2024-10-28T21:27:20.682+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
2024-10-28T21:27:20.688+0200	INFO	dractions/actions.go:175	update drpc subscr-deploy-cephfs-busybox failover to dr2
2024-10-28T21:27:55.707+0200	INFO	dractions/retry.go:114	drpc subscr-deploy-cephfs-busybox phase is FailedOver
2024-10-28T21:28:55.717+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Relocate
2024-10-28T21:28:55.717+0200	INFO	dractions/actions.go:130	enter Relocate subscr-deploy-cephfs-busybox
2024-10-28T21:28:55.719+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
2024-10-28T21:28:55.725+0200	INFO	dractions/actions.go:175	update drpc subscr-deploy-cephfs-busybox relocate to dr1
2024-10-28T21:31:00.789+0200	INFO	dractions/retry.go:114	drpc subscr-deploy-cephfs-busybox phase is Relocated
2024-10-28T21:31:30.800+0200	INFO	dractions/retry.go:57	drpc subscr-deploy-cephfs-busybox is ready
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Disable
2024-10-28T21:31:30.801+0200	INFO	dractions/actions.go:99	enter DisableProtection subscr-deploy-cephfs-busybox
2024-10-28T21:31:30.801+0200	INFO	dractions/actions.go:104	delete drpc subscr-deploy-cephfs-busybox
2024-10-28T21:32:05.823+0200	INFO	dractions/retry.go:183	drpc subscr-deploy-cephfs-busybox is deleted
=== RUN   TestSuites/Exhaustive/Deploy-cephfs/Subscr/Undeploy
2024-10-28T21:32:05.824+0200	INFO	deployers/subscription.go:62	enter Undeploy subscr-deploy-cephfs-busybox
2024-10-28T21:32:05.927+0200	INFO	util/crud.go:99	channel ramen-gitops is deleted
--- PASS: TestSuites (0.00s)
    --- PASS: TestSuites/Exhaustive (0.07s)
        --- PASS: TestSuites/Exhaustive/Deploy-cephfs#02 (0.00s)
        --- PASS: TestSuites/Exhaustive/Deploy-cephfs#01 (0.00s)
        --- PASS: TestSuites/Exhaustive/Deploy-cephfs (0.00s)
            --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr (380.33s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Deploy (5.05s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Enable (90.08s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Failover (95.04s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Relocate (155.06s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Disable (35.02s)
                --- PASS: TestSuites/Exhaustive/Deploy-cephfs/Subscr/Undeploy (0.07s)
PASS
ok  	github.com/ramendr/ramen/e2e	381.044s
```

--- a/e2e/dractions/actions.go
+++ b/e2e/dractions/actions.go
@@ -83,8 +83,8 @@ func EnableProtection(w workloads.Workload, d deployers.Deployer) error {
 	}
 
 	// this is the application namespace in drclusters to add the annotation
-	nsToAnnonate := name
-	if err := util.CreateNamespaceAndAddAnnotation(nsToAnnonate); err != nil {
+	nsToAnnotate := name
+	if err := util.CreateNamespaceAndAddAnnotation(nsToAnnotate); err != nil {
 		return err
 	}
 

--- a/e2e/util/crud.go
+++ b/e2e/util/crud.go
@@ -5,6 +5,7 @@ package util
 
 import (
 	"context"
+	"fmt"
 	"time"
 
 	"k8s.io/apimachinery/pkg/api/errors"
@@ -55,11 +56,11 @@ func DeleteNamespace(client client.Client, namespace string) error {
 	}
 
 	Ctx.Log.Info("waiting until namespace " + namespace + " is deleted")
+
+	startTime := time.Now()
 	key := types.NamespacedName{Name: namespace}
 
 	for {
-		time.Sleep(time.Second)
-
 		if err := client.Get(context.Background(), key, ns); err != nil {
 			if !errors.IsNotFound(err) {
 				return err
@@ -69,6 +70,12 @@ func DeleteNamespace(client client.Client, namespace string) error {
 
 			return nil
 		}
+
+		if time.Since(startTime) > 60*time.Second {
+			return fmt.Errorf("timeout deleting namespace %q", namespace)
+		}
+
+		time.Sleep(time.Second)
 	}
 }
 

--- a/e2e/util/crud.go
+++ b/e2e/util/crud.go
@@ -5,6 +5,7 @@ package util
 
 import (
 	"context"
+	"time"
 
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
@@ -49,9 +50,26 @@ func DeleteNamespace(client client.Client, namespace string) error {
 		}
 
 		Ctx.Log.Info("namespace " + namespace + " not found")
+
+		return nil
 	}
 
-	return nil
+	Ctx.Log.Info("waiting until namespace " + namespace + " is deleted")
+	key := types.NamespacedName{Name: namespace}
+
+	for {
+		time.Sleep(time.Second)
+
+		if err := client.Get(context.Background(), key, ns); err != nil {
+			if !errors.IsNotFound(err) {
+				return err
+			}
+
+			Ctx.Log.Info("namespace " + namespace + " deleted")
+
+			return nil
+		}
+	}
 }
 
 func createChannel() error {

